src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -240,6 +250,50 @@ private void transferRoutes(){
                     e.getIn().setBody(objectMapper.writeValueAsString(response));
                 });
 
+        //fetch transfer details based on client correaltion id
+        from("rest:GET:/channel/txnState/{correlationId}")
+                .id("transfer-details-correlationid")
+                .log(LoggingLevel.INFO, "## CHANNEL -> inbound transferDetail request for client correlation Id" +
+                        " ${header.correlationId}")
+                .process(e -> {
+                    String tenantId = e.getIn().getHeader("Platform-TenantId", String.class);
+                    if (tenantId == null || !dfspIds.contains(tenantId)) {
+                        throw new RuntimeException("Requested tenant " + tenantId + " not configured in the connector!");}
+                    Client client = clientProperties.getClient(tenantId);
+                    String requestType = e.getIn().getHeader("requestType", String.class);
+                    requestType = requestType.isEmpty() ? "transfers" : requestType;
+                    requestType = requestType.equalsIgnoreCase("transfers") ? transfersEndpoint :
+                            transactionEndpoint;
+                    HttpEntity<String> entity = buildHeader(tenantId, null);
+                    UriComponentsBuilder builder = buildParams(client);
+                    ResponseEntity<String> exchange = restTemplate.exchange(builder.toUriString(), HttpMethod.POST,
+                            entity, String.class);
+                    JSONObject jsonObject = new JSONObject(exchange.getBody());
+                    String token = jsonObject.getString("access_token");
+                    logger.info("Got token {}",token);
+                    String correlationId = e.getIn().getHeader("correlationId", String.class);
+                    entity = buildHeader(tenantId, token);
+                    logger.info("Final url: {}", operationsUrl + requestType +"clientCorrelationId=" +
+                            correlationId);
+                    exchange = restTemplate.exchange(operationsUrl + requestType +"clientCorrelationId=" +
```

Comment:
1. use string format instead of + here
2. same comment as above for rest tempate